daffodil/base_delegate.py

```diff
@@ -1,29 +1 @@
-from builtins import object
-
-
-class BaseDaffodilDelegate(object):
-
-    def mk_any(self, children):
-        raise NotImplementedError()
-
-    def mk_all(self, children):
-        raise NotImplementedError()
-
-    def mk_not_any(self, children):
-        raise NotImplementedError()
-
-    def mk_not_all(self, children):
-        raise NotImplementedError()
-
-    def mk_test(self, test_str):
-        raise NotImplementedError()
-
-    def mk_comment(self, comment, is_inline):
-        raise NotImplementedError()
-
-    def mk_cmp(self, key, val, test):
-        raise NotImplementedError()
-
-    def call(self, predicate, iterable):
-        raise NotImplementedError()
-
+from .parser import BaseDaffodilDelegate
```

Comment:
This is for backwards compat - I move `BaseDaffodilDelegate` to parser.py to avoid a particularly pernicious cyclical import issue that was affecting the generated C - code (which suffers a performance penalty if you do fancy stuff like importing within a function) and doesn't have python's fancy import system.